bundles/org.openhab.binding.ojelectronics/src/main/java/org/openhab/binding/ojelectronics/internal/services/RefreshService.java

```diff
@@ -0,0 +1,158 @@
+import org.openhab.binding.ojelectronics.internal.models.groups.GroupContentResponseModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Handles the refreshing of the devices of a OJElectronics session
+ *
+ * @author Christian Kittel - Initial Contribution
+ */
+@NonNullByDefault
+public final class RefreshService implements AutoCloseable {
+
+    private final OJElectronicsBridgeConfiguration config;
+    private final Logger logger = LoggerFactory.getLogger(RefreshService.class);
+    private final HttpClient httpClient;
+    private final Gson gson = new GsonBuilder().setPrettyPrinting().setDateFormat("yyyy-MM-dd'T'HH:mm:ss").create();
+    private final ScheduledExecutorService schedulerService;
+
+    private @Nullable Runnable connectionLost;
+    private @Nullable Consumer<@Nullable GroupContentResponseModel> refreshDone;
+    private @Nullable ScheduledFuture<?> scheduler;
+    private @Nullable Runnable unauthorized;
+    private String sessionId = "";
+    private static boolean destroyed = false;
+
+    /**
+     * Creates a new instance of {@link RefreshService}
+     *
+     * @param config Configuration of the bridge
+     * @param httpClient HTTP client
+     */
+    public RefreshService(OJElectronicsBridgeConfiguration config, HttpClient httpClient,
+            ScheduledExecutorService schedulerService) {
+        this.config = config;
+        this.httpClient = httpClient;
+        this.schedulerService = schedulerService;
+    }
+
+    /**
+     * Starts refreshing all thing values
+     *
+     * @param sessionId Session-Id
+     * @param refreshDone This method is called if refreshing is done.
+     * @param connectionLosed This method is called if no connection could established.
+     * @param unauthorized This method is called if the result is unauthorized.
+     */
+    public void start(String sessionId, Consumer<@Nullable GroupContentResponseModel> refreshDone,
+            Runnable connectionLost, Runnable unauthorized) {
+        logger.trace("RefreshService.startService({})", sessionId);
+        this.connectionLost = connectionLost;
+        this.refreshDone = refreshDone;
+        this.unauthorized = unauthorized;
+        this.sessionId = sessionId;
+        long refreshTime = config.refreshDelayInSeconds;
+        scheduler = schedulerService.scheduleWithFixedDelay(this::refresh, refreshTime, refreshTime, TimeUnit.SECONDS);
+        destroyed = false;
+    }
+
+    /**
+     * Stops refreshing.
+     */
+    public void stop() {
+        destroyed = true;
+        final ScheduledFuture<?> scheduler = this.scheduler;
+        if (scheduler != null) {
+            scheduler.cancel(false);
+        }
+        this.scheduler = null;
+    }
+
+    private void refresh() {
+        if (sessionId.equals("")) {
```

Comment:
It might make more sense to just make this fields nullable, it seems you have "" a magical meaning instead.

If you want to leave, then position the literal first in `String` comparisons - that way if the `String` is `null` you won't get a `NullPointerException`, it'll just return `false`. [source: PMD](http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#PositionLiteralsFirstInComparisons)

Reply:
Done